sklearn/preprocessing/_polynomial.py

```diff
@@ -1052,6 +1118,10 @@ def transform(self, X):
                     and np.any(
                         np.isnan(XBS[:, (i * n_splines) : ((i + 1) * n_splines)])
                     )
+                    # to distinguish original np.nan values from the ones
+                    # created by any element in X < xmin (=spl.t[degree]) or >
+                    # xmax (=spl.t[-degree - 1])
+                    and np.any((X < spl.t[degree]) | (X > spl.t[-degree - 1]))
```

Comment:
I believe this is no longer needed now that always replace nan via valid, in-range input values (`np.nanmin`).

```suggestion
```